tree/ntuple/v7/inc/ROOT/RNTupleMerger.hxx

```diff
@@ -46,50 +66,16 @@ struct RNTupleMergeOptions {
  *        This can also be used to change the compression of a single RNTuple by just passing a single source.
  */
 // clang-format on
-class RNTupleMerger {
-
-private:
-   // Struct to hold column information
-   struct RColumnInfo {
-      std::string fColumnName; ///< The qualified field name to which the column belongs, followed by the column index
-      std::string fColumnTypeAndVersion; ///< "<type>.<version>" of the field to which the column belongs
-      DescriptorId_t fColumnInputId;
-      DescriptorId_t fColumnOutputId;
-
-      RColumnInfo(const std::string &name, const std::string &typeAndVersion, const DescriptorId_t &inputId,
-                  const DescriptorId_t &outputId)
-         : fColumnName(name), fColumnTypeAndVersion(typeAndVersion), fColumnInputId(inputId), fColumnOutputId(outputId)
-      {
-      }
-   };
-
-   /// Build the internal column id map from the first source
-   /// This is where we assign the output ids for the first source
-   void BuildColumnIdMap(std::vector<RColumnInfo> &columns);
-
-   /// Validate the columns against the internal map that is built from the first source
-   /// This is where we assign the output ids for the remaining sources
-   void ValidateColumns(std::vector<RColumnInfo> &columns);
-
-   /// Recursively add columns from a given field
-   void AddColumnsFromField(std::vector<RColumnInfo> &columns, const RNTupleDescriptor &desc,
-                            const RFieldDescriptor &fieldDesc, const std::string &prefix = "");
-
-   /// Recursively collect all the columns for all the fields rooted at field zero
-   void CollectColumns(const RNTupleDescriptor &descriptor, std::vector<RColumnInfo> &columns);
-
-   // Internal map that holds column name, type, and type id : output ID information
-   std::unordered_map<std::string, DescriptorId_t> fOutputIdMap;
-
+class RNTupleMerger final {
 public:
+   struct RChangeCompressionFunc;
```

Comment:
It is currently needed for the `friend` access (the whole reason to have friend access is being able to access `UnsealPage` from therein). I'm not sure how to avoid it, except by making UnsealPage public of course...